File: submitted code/100329.zai-pai-xu-shu-zu-zhong-cha-zhao-shu-zi-lcof.java
//
// @lc app=leetcode.cn id=100329 lang=java
//
// [100329] zai-pai-xu-shu-zu-zhong-cha-zhao-shu-zi-lcof
//
class Solution {
    public int search(int[] nums, int target) {
        if(nums.length == 0)
            return 0;
        int l = 0, r = nums.length -1;
        while(l < r){
            int mid = (l+r)/2;
            if(nums[mid] < target)
                l = mid + 1;
            else
                r = mid;
        }
        if(nums[l] != target)
            return 0;
        int a = l;
        l = 0;
        r = nums.length-1;
        while(l < r){
            int mid = (l+r+1)/2;
            if(nums[mid] > target)
                r = mid - 1;
            else
                l = mid;
        }
        return l - a + 1;
    }
}
// @lc code=end

File: submitted code/100329.zai-pai-xu-shu-zu-zhong-cha-zhao-shu-zi-lcof.cpp
//
// @lc app=leetcode.cn id=100329 lang=cpp
//
// [100329] zai-pai-xu-shu-zu-zhong-cha-zhao-shu-zi-lcof
//
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int len = nums.size();
        if(len == 0 || target < nums[0] || target > nums[len-1])
            return 0;
        int l = 0, r= len-1;
        while(l < r){
            int m = (l+r)/2;
            if(nums[m] >= target)
                r = m;
            else
                l = m+1;
        }
        //System.out.println(nums[l]);
        int t = l;
        l = 0; r = len-1;
        while(l < r){
            int m = (l+r+1)/2;
            if(nums[m] <= target)
                l = m;
            else
                r = m-1;
        }
        //System.out.println(nums[l]);
        if(nums[l] == target && nums[t] == target)
            return l - t + 1;
        if(l == t || nums[t] == target || nums[l] == target)
            return 1;
        return 0;
    }
};
// @lc code=end